### ISSUE TITLE
Create alternative to std_msgs/Float64MultiArray in ForwardCommandController & family

### ISSUE THREAD

**Issue by bmagyar** [ros-controls/ros2_controllers#157]
# Problem:
std_msgs/Float64MultiArray is deprecated.

```
# This was originally provided as an example message.
# It is deprecated as of Foxy
# It is recommended to create your own semantically meaningful message.
# However if you would like to continue using this please use the equivalent in example_msgs.

# Please look at the MultiArrayLayout message definition for
# documentation on all multiarrays.

MultiArrayLayout  layout        # specification of data layout
float64[]         data          # array of data
```

https://github.com/ros2/common_interfaces/blob/foxy/std_msgs/msg/Float64MultiArray.msg

# Proposal:
Create (an) alternative in `control_msgs`. This could be a set of messages with a single joint interface type, one supporting multiples, or one message that has a field for the joint interface type and one that is fully dynamic --- (dynamic joint states??).

**Comment by gavanderhoorn** [comment]
> std_msgs/Float64MultiArray is deprecated.

yeah, this really should have been given some more attention.

See [SOSS: a whole new approach to your ROS 1-ROS 2 bridge!](https://discourse.ros.org/t/soss-a-whole-new-approach-to-your-ros-1-ros-2-bridge/17040/8) where this was casually dropped in an off-topic discussion. "It's in the release notes, we can't do anything more" (which is true, it's mentioned in [the notes](https://docs.ros.org/en/foxy/Releases/Release-Foxy-Fitzroy.html#std-msgs), but I doubt many people have seen those).

**Comment by bmagyar** [comment]
Agreed. I don't think I've ever read the release notes of a ros distro 😓

**Comment by destogl** [comment]
Yeah, I also had no idea, and although I looked at this message a few times, I didn't read its header. Actually, a ROS newcomer I was helping pointed me into this...

This should be a massive topic on ROS discourse...

**Comment by gavanderhoorn** [comment]
Feel free to bring it up again.

I've directly asked for an announcement. The response was:

> It's in the release notes, we can't do anything more

**Comment by ijnek** [comment]
Mentioned in [Discourse Post](https://discourse.ros.org/t/msg-type-for-publishing-jointcommands/22451)

**Comment by RhysMcK** [comment]
Any update on this?

**Comment by bmagyar** [comment]
Well it's deprecated but still around... If you want to introduce something for this yourself you are welcome to submit a PR to `control_msgs`

**Comment by BrittanyKCohen96** [comment]
Hi, I'm a 4th year EE student with 1YOE in ROS2 (from working on my final project). I found this while searching for a good first issue to work on. I've used these messages to post multiple joint positions for the JointGroupPositionController so I have some experience with them. Is this something that someone with 1YOE can work on?

**Comment by christophfroehlich** [comment]
You are welcome to contribute 🎉
First step is a proposal for a new message in control_msgs repository. We thought of something similar to https://github.com/ros2/common_interfaces/blob/rolling/trajectory_msgs/msg/JointTrajectory.msg and https://github.com/ros2/common_interfaces/blob/rolling/trajectory_msgs/msg/JointTrajectoryPoint.msg
No trajectory, only a single point but where one can also send a command only to a single interface (defined by the name field; this logic is not implemented in the forward_command_controller yet).

**Comment by BrittanyKCohen96** [comment]
Thank you! This sounds really good, that was exactly the issue that we had when we were using it. I'm in final exams but I'll get started on the 19th of August.

**Comment by christophfroehlich** [comment]
Fine. The message itself is little work but will need some time for discussion and review rounds ;) just in case you've got a minute for a proposal

**Comment by BrittanyKCohen96** [comment]
Hi, I've been waiting for confirmation from the company that I work for and it still hasn't arrived, but it looks like working on ROS2 in general is a conflict of interest. Still waiting on confirmation but thought I'd let you know ASAP so the task can be reassigned. I'm really sorry for the inconvenience.

**Comment by ShahazadAbdulla** [comment]
Hi @bmagyar and @christophfroehlich , since this was opened up again in September, I would like to pick this up. I would like to work on this as @BrittanyKCohen96 have stated that the task can be reassigned and as I understood from the previous messages I will draft a proposal for the new message in control_msgs as outlined above and link it here shortly.

**Comment by ShahazadAbdulla** [comment]
I have opened a PR in control_msgs proposing the JointCommand.msg structure to provide the semantically meaningful, timestamped alternative discussed here.

The new msg is named JointCommand.msg and the structure is:
``` bash
root@1f21ca7190a1:/ros2_ws# ros2 interface show control_msgs/msg/JointCommand
# This message contains the command for a specific interface across multiple joints.

std_msgs/Header header
        builtin_interfaces/Time stamp
                int32 sec
                uint32 nanosec
        string frame_id

# List of joint names to command.
string[] joint_names

# The specific interface type to command (e.g., "position", "velocity", "effort").
string interface_name

# The commanded values. The length of this array must exactly match the length of joint_names.
float64[] values
```

PR: ros-controls/control_msgs#304

**Comment by christophfroehlich** [comment]
@ShahazadAbdulla do you want to take care of adding the new command topic to forward_command_controller and update the demos? We should not remove the current topic, but deprecate it first. Please evaluate if a node can create two subscribers with the same topic but different message type. If this is not easily possible, propose a new one instead of `commands`

**Comment by ShahazadAbdulla** [comment]
@christophfroehlich Yes I will take of the integration and demo updates. As DDS doesnt support this, so a node cant make two subscribers with different msg types on the same topic name. 

How about as you said lets deprecate the Float64MultiArray subscriber to the topic '/command'/ and RCL warn about the deprecation of this msg type. And then spin up a new topic named '/joint_command' with our JointCommand msg type. We could have a dual subscriber kind and add some fall back handlers and all. How about this. Any other suggestions.

**Comment by christophfroehlich** [comment]
> As DDS doesnt support this, so a node cant make two subscribers with different msg types on the same topic name.

are you sure? At least there should be something like GenericSubscription
